02.calendar/cal.rb

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+require 'optparse'
+require 'date'
+
+options = ARGV.getopts('y:', 'm:')
+
+input_year = options["y"]
+input_month = options["m"]
+
+# -yが入力されなかった場合は今年を指定
+target_year = input_year.nil? ? Date.today.year : input_year.to_i
+# -mが入力されなかった場合は今月を指定
+target_month = input_month.nil? ? Date.today.month : input_month.to_i
+
+# 対象月の初日
+first_date_of_month = Date.new(target_year, target_month, 1)
+# 対象月の最終日
+last_date_of_month = Date.new(target_year, target_month, -1)
+
+# 表示用の対象月('January' ~ 'December')
+display_month = first_date_of_month.strftime('%B')
+# 表示用の省略形の全曜日
+display_days = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'].join(' ')
```

Comment:
```suggestion
display_days = "Su Mo ... Sa"
```

のように単純に1つの文字列にした方がシンプルになりそうです